### PR TITLE
Fixed #22654 -- Broken decimal validation

### PR DESCRIPTION
**Pull request by raphaelm**
see https://code.djangoproject.com/ticket/22654

### CODE REVIEW

**Comment by apollo13** [comment]
I am trying to understand the implementation of `get_format` currently, it looks as if it should fall back to settings if L10N is not enabled (apparently buggy in that case)

**Comment by raphaelm** [comment]
> I am trying to understand the implementation of get_format currently, it looks as if it should fall back to settings if L10N is not enabled (apparently buggy in that case)

Oh, yes, ``get_format`` would work probably. I'll try. It just hasn't been called before ;)

File: docs/releases/2.0.txt [hunk 1]
```diff
@@ -157,6 +157,10 @@ Forms
   HTML attributes for the ``DateInput`` and ``TimeInput`` (or hidden)
   subwidgets.
 
+* :class:`~django.forms.FloatField` and :class:`~django.forms.DecimalField`
```
Comment (timgraham) [inline docs/releases/2.0.txt]: This looks more like a bug fix than a "minor feature" so we wouldn't have a release note for that typically.

**Comment by raphaelm** [comment]
Hi @timgraham,

> Did you consider some tests for in tests/i18n/tests.py? (test_sanitize_separators). I'm not sure if tests should be repeated everywhere sanitize_separators() is used (i.e. in both fields) but I guess it's okay.


I just didn't think of ``test/i18n``, as the bug was reported against the form field and I wrote the regression test before I looked at the implementation :wink: Would be a better fit, I guess. 

> For future reference, try to use a more descriptive commit message, something like "Made FloatField/DecimalField handle DECIMAL_SEPARATOR and THOUSAND_SEPARATOR when USE_L10N=False".

I wasn't happy with the commit message myself, but had real trouble coming with a short enough subject line …

> This looks more like a bug fix than a "minor feature" so we wouldn't have a release note for that typically.

Seems legit. Do you consider those things (the release notes and the tests in a different place) worthy for a new PR, as this is already merged now?

**Comment by timgraham** [comment]
Yes, it would be nice to address my comments as you see fit.

**Comment by raphaelm** [comment]
Okay, will do so. Shall I keep the "duplicate" tests or get rid of them?

**Comment by timgraham** [comment]
I suppose they don't hurt.

**Comment by raphaelm** [comment]
Okay, see you at #8365 :)

**Pull request merged**